### ISSUE TITLE
Quelpa should wait for the cache file to be unlocked

### ISSUE THREAD

**Issue by DarwinAwardWinner** [quelpa/quelpa#147]
If multiple copies of Emacs are starting up at once, they will fight over the cache file. This problem occurred for me when I was configuring an org-mode file to export to multiple formats asynchronously upon each save. Each asynchronous export job spawns a new Emacs process, which loads the init file and therefore runs Quelpa. One of the processes hangs because it asks the user if it should steal the cache file from the other, but since this is a subprocess being run by org-mode's export functionality, there is no way to interact with it. Instead, if Emacs is running in batch mode, I think Quelpa should override this interactive lock-handling to simply wait until the file is unlocked and then continue normally.

**Comment by wasamasa** [comment]
Sorry, but this sounds like a bad idea to me. It enables deadlocks and what's currently happening is in line with other package managers that exit early when encountering locked resources.

Besides this, is there a need to run Quelpa during a batch operation? Why not make it conditional and not run it in batch mode?

**Comment by DarwinAwardWinner** [comment]
Well, Quelpa is pretty baked in to my Emacs config, so it might not be easy to make it conditional. However, I did discover an alternate solution for my specific case: org-mode's export functions allow you to specify a different init file to use for the asynchronous Emacs subprocesses in place of the usual one, so I was able to craft a separate init file that does not use Quelpa. So I don't have a strong need for this any more.
